web_src/js/features/comp/WebHookEditor.js

```diff
@@ -1,11 +1,69 @@
 import $ from 'jquery';
 const {csrfToken} = window.config;
 
+const initAuthenticationHeaderSection = function() {
+  const $authHeaderSection = $('.auth-headers');
+
+  if ($authHeaderSection.length === 0) {
+    return;
+  }
+
+  const $checkbox = $authHeaderSection.find('.checkbox input');
+
+  const updateHeaderContentType = function() {
+    const isBasicAuth = $authHeaderSection.find('#auth_header_type').val() === 'basic';
+    const $basicAuthFields = $authHeaderSection.find('.basic-auth');
+    const $tokenAuthFields = $authHeaderSection.find('.token-auth');
+
+    if (isBasicAuth) {
+      $basicAuthFields.addClass('required');
+      $basicAuthFields.find('input').attr('required', '');
+      $basicAuthFields.show();
+
+      $tokenAuthFields.removeClass('required');
+      $tokenAuthFields.find('input').removeAttr('required');
+      $tokenAuthFields.hide();
+    } else {
+      $basicAuthFields.removeClass('required');
+      $basicAuthFields.find('input').removeAttr('required');
+      $basicAuthFields.hide();
+
+      $tokenAuthFields.addClass('required');
+      $tokenAuthFields.find('input').attr('required', '');
+      $tokenAuthFields.show();
+    }
+  };
+
+  const updateHeaderCheckbox = function() {
+    if ($checkbox.is(':checked')) {
+      const $headerName = $authHeaderSection.find('#auth_header_name');
+      $headerName.attr('required', '');
+      $headerName.parent().addClass('required');
+      $headerName.parent().show();
```

Comment:
```suggestion
      $headerName.parent().addClass('required').show();
```

This and similar cases can be chained for performance and conciseness.